apps/sms/js/thread_list_ui.js

```diff
@@ -857,13 +846,7 @@ var ThreadListUI = {
     var threadUITime = threadUINode ? +threadUINode.dataset.time : NaN;
     var recordTime = +thread.timestamp;
 
-    // For legitimate in-memory thread objects, update the stored
-    // Thread instance with the newest data. This check prevents
-    // draft objects from inadvertently creating bogus thread
-    // objects.
-    if (Threads.has(thread.id)) {
-      Threads.set(thread.id, thread);
-    }
+    Threads.set(thread.id, thread);
```

Comment:
I do exactly the same here: https://github.com/mozilla-b2g/gaia/pull/28681/files#diff-e2424773c46c9d5f4ff27c9c20b1bb0dL851

(I changed the name of Threads.set).

So ok for me :D